### DISCUSSION TITLE
Client not sending SID and instead sending no data at all

### DISCUSSION THREAD

**Discussion by apknusel** [miguelgrinberg/python-socketio#1356] [Q&A]
I am trying to connect to a WebSocket server, and instead of the client sending a `40{"sid":"Umj7HNgYWvJ3AU8RAHrR"}`, it is sending a 0{}. Not entirely why this is but I do not think this is intended behavior. I know I have the proper headers and cookies because I was able to connect using Postman with the same headers and cookies. Line of code causing the error: `sio.connect(url="wss://REDACTED/socket.io/?b64=1", transports=['websocket'], headers=headers)`

```Attempting WebSocket connection to wss://REDACTED/socket.io/?b64=1&transport=websocket&EIO=4
WebSocket connection accepted with {'sid': 'Umj7HNgYWvJ3AU8RAHrR', 'upgrades': [], 'pingInterval': 25000, 'pingTimeout': 20000, 'maxPayload': 1000000}
Engine.IO connection established
Sending packet MESSAGE data 0{}
Sending packet CLOSE data None
Engine.IO connection dropped
Exiting write loop task
WebSocket connection was closed, aborting
Waiting for write loop task to end
Exiting read loop task```

**Comment by miguelgrinberg** [comment]
The argument that goes in the `sio.connect()` call is a domain, you are providing a URL that points to the Socket.IO endpoint.

I recommend you try some of the example servers and clients in this repository to learn how this package works instead of paying attention to the wire. The intention is to hide the details of the protocol as much as possible.

Also, you refer to some error, but you haven't indicated what the error is. Please provide a stack trace. The log that you shared indicates that the client is closing the connection immediately after opening it.

**Reply by apknusel** [reply]
I figured out my issue. It was all user error, my apologies.
